tools/buildsys/src/builder.rs

```diff
@@ -44,10 +45,23 @@ pub(crate) struct PackageBuilder;
 
 impl PackageBuilder {
     /// Build RPMs for the specified package.
-    pub(crate) fn build(package: &str) -> Result<Self> {
+    pub(crate) fn build(
+        package: &str,
+        supported_arches: Option<&HashSet<SupportedArch>>,
+    ) -> Result<Self> {
         let arch = getenv("BUILDSYS_ARCH")?;
         let output_dir: PathBuf = getenv("BUILDSYS_PACKAGES_DIR")?.into();
 
+        if let Some(supported_arches) = supported_arches {
+            let current_arch: SupportedArch =
+                serde_plain::from_str(&arch).context(error::UnknownArch { arch: &arch })?;
+
+            ensure!(
+                supported_arches.contains(&current_arch),
+                error::UnsupportedArch { arch: &arch }
+            )
+        }
+
```

Comment:
Let's extract this into a helper function in `main.rs` that we can call from `build_package` or `build_variant` before we get to this point. Otherwise we're complicating the function signature and duplicating the logic here.